modules/core/src/loaders/xviz-loader-interface.js

```diff
@@ -253,10 +265,20 @@ export default class XVIZLoaderInterface {
   }
 
   _onXVIZTimeslice(timeslice) {
+    const oldStreamCount = this.streamBuffer.streamCount;
     const bufferUpdated = this.streamBuffer.insert(timeslice);
     if (bufferUpdated) {
       this._bumpDataVersion();
     }
+
+    if (getXVIZConfig().DYNAMIC_STREAM_METADATA && this.streamBuffer.streamCount > oldStreamCount) {
```

Comment:
the second part of the if statement might be too lax (`this.streamBuffer.streamCount > oldStreamCount`). We could have the same number of streams, but different stream names. And in this case, we would fail to update `streamsMetadata`.

E.g:
```
{
   `/vehicle/pose`: ...,
   `/lidar`: ...
}
```

```
{
   `/vehicle/speed`: ...,
   `/lidar`: ...
}
```

I can see how doing a stream by stream comparison can be expensive one very update, so maybe we want to check on the exact `streamNames` list only periodically (every 1s?). Just an idea to not check this on every update.

Reply:
`streamCount` is an accumulative number since the creation of the buffer, and it only ever increases.